### PR TITLE
support cli for generate schema and fix bugs

### PR DESCRIPTION
**Pull request by aiwantaozi**
**Problem:**
1. Bug: Could not import template include option(type, default)
2. Bug: Need to support the template only define default without type
3. Bug: Sync template from catalog renders incorrect schema
4. Support CLI for generating schema.yaml 

**Solution:**
1. Support import template with option(type, default)
2. Support generate type from default
3. Skip the incorrectly cached schema while syncing templates from catalog
4. Support CLI command ui-schema generate

**Related Issue:**
https://github.com/seal-io/walrus/issues/1355
https://github.com/seal-io/walrus/issues/1362
#1365

### CODE REVIEW

File: pkg/templates/loader/loader.go [hunk 1]
```diff
@@ -8,15 +8,31 @@ import (
 
 // SchemaLoader define the interface for loading schema from template.
 type SchemaLoader interface {
-	Load(rootDir, templateName, templateVersion string) (*types.Schema, error)
+	Load(rootDir, templateName string, mode Mode) (*types.Schema, error)
```
Comment (gitlawr) [inline pkg/templates/loader/loader.go:11]: Whether it's from schema file or not is the implementation detail of SchemaLoader. Why do we expose it to the caller

Comment (aiwantaozi) [reply pkg/templates/loader/loader.go:11]: Expose to support different modes.
- Load from file: mainly used in walrus server. While walrus sync template schema, when there is a schema.yaml, will use the schema user defined in schema.yaml directly.
- Merge: mainly used in CLI, while user already defined a customized schema.yaml, and then updated the varaibles.tf this mode will regenerate the schema from variables.tf and merge the result with the schema from schema.yaml, this mode will include both new generated and user customized.
- Original: only return the generated, ignore the customized.

File: cmd/cli/command.go [hunk 2]
```diff
@@ -146,6 +154,50 @@ func NewConfigCmd() *cobra.Command {
 	return configCmd
 }
 
+// NewUISchemaCmd generate ui schema command.
+func NewUISchemaCmd() *cobra.Command {
```
Comment (gitlawr) [inline cmd/cli/command.go:158]: Why UISchema instead of Schema?

Comment (aiwantaozi) [reply cmd/cli/command.go]: updated

File: pkg/templates/loader/loader.go [hunk 3]
```diff
@@ -26,5 +42,13 @@ func LoadSchema(rootDir, templateName, templateVersion string) (s *types.Templat
...
 	// Continue with other loaders in the future.
-	return nil, fmt.Errorf("no supported schema found for template %s version %s", templateName, templateVersion)
+	return nil, fmt.Errorf("no supported schema found for template %s", templateName)
 }
+
+type Mode string
+
+const (
+	ModePreferFromFile Mode = "preferFromFile"
+	ModeFromOriginal   Mode = "fromOriginal"
```
Comment (gitlawr) [inline pkg/templates/loader/loader.go:52]:
```
ModeSchemaFile
ModeOriginal
ModeMerge
```

**Review by gitlawr — DISMISSED** [review]
LGTM

File: pkg/templates/loader/loader.go [hunk 4]
```diff
@@ -26,5 +42,13 @@ func LoadSchema(rootDir, templateName, templateVersion string) (s *types.Templat
 	}
 
 	// Continue with other loaders in the future.
-	return nil, fmt.Errorf("no supported schema found for template %s version %s", templateName, templateVersion)
+	return nil, fmt.Errorf("no supported schema found for template %s", templateName)
 }
+
+type Mode string
+
+const (
```
Comment (alexcodelf) [inline pkg/templates/loader/loader.go:50]: can we make the mode define style as same as https://github.com/seal-io/walrus/blob/main/pkg/apis/server.go#L52-L56?

**Review by alexcodelf — APPROVED** [review]
LGTM

**Pull request merged**